src/Materials/Textures/babylon.colorGradingTexture.ts

```diff
@@ -86,12 +86,12 @@ module BABYLON {
 
             this._texture = texture;
 
-            var callback = (text: string) => {
+            var callback = (text: string | ArrayBuffer) => {
                 var data: Nullable<Uint8Array> = null;
                 var tempData: Nullable<Float32Array> = null;
 
                 var line: string;
-                var lines = text.split('\n');
+                var lines = (<string>text).split('\n');

```

Comment:
What happens if text is an ArrayBuffer? we can test here `if(typeof text ==='string')` which will remove the ts error and will actually be safer :)

Reply:
fixed

Reply:
Now you dont need the casting

Reply:
Sure? I think that TS will still enforce it

Reply:
Excellent it works
I learn something today :D